src/Features/Core/Portable/ConflictMarkerResolution/AbstractConflictMarkerCodeFixProvider.cs

```diff
@@ -109,27 +123,66 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
 
         private static bool TryGetConflictLines(
             SourceText text, int position,
-            out TextLine startLine, out TextLine middleLine, out TextLine endLine)
+            out TextLine startLine, out TextLine firstMiddleLine, out TextLine secondMiddleLine, out TextLine endLine)
         {
             startLine = default;
-            middleLine = default;
+            firstMiddleLine = default;
+            secondMiddleLine = default;
             endLine = default;
 
             var lines = text.Lines;
+            bool foundBarLine;
+            bool success;
             switch (text[position])
             {
                 case '<':
                     startLine = lines.GetLineFromPosition(position);
-                    return TryFindLineForwards(startLine, '=', out middleLine) &&
-                           TryFindLineForwards(middleLine, '>', out endLine);
+                    foundBarLine = TryFindLineForwards(startLine, '|', out firstMiddleLine);
+
+                    success = TryFindLineForwards(foundBarLine ? firstMiddleLine : startLine, '=', out secondMiddleLine) &&
+                              TryFindLineForwards(secondMiddleLine, '>', out endLine);
+
+                    if (success && !foundBarLine)
+                    {
+                        firstMiddleLine = secondMiddleLine;
+                    }
+
+                    return success;
+                case '|':
+                    firstMiddleLine = lines.GetLineFromPosition(position);
+                    return TryFindLineBackwards(firstMiddleLine, '<', out startLine) &&
+                           TryFindLineForwards(firstMiddleLine, '=', out secondMiddleLine) &&
```

Comment:
ok.  so if you have `||||||||||` you must have `==========`?

Reply:
Yes. Addressed by adding some doc at top of the file